server/src/main/java/org/opensearch/action/support/replication/FanoutReplicationProxy.java

```diff
@@ -8,15 +8,46 @@
 
 package org.opensearch.action.support.replication;
 
+import org.opensearch.action.ActionListener;
 import org.opensearch.cluster.routing.ShardRouting;
 
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
 /**
  * This implementation of {@link ReplicationProxy} fans out the replication request to current shard routing if
  * it is not the primary and has replication mode as {@link ReplicationMode#FULL_REPLICATION}.
  *
  * @opensearch.internal
  */
-public class FanoutReplicationProxy<ReplicaRequest> extends ReplicationProxy<ReplicaRequest> {
+public class FanoutReplicationProxy<ReplicaRequest extends ReplicationRequest<ReplicaRequest>> extends ReplicationProxy<ReplicaRequest> {
+
+    public FanoutReplicationProxy(ReplicationOperation.Replicas<ReplicaRequest> replicasProxy) {
+        super(replicasProxy);
+    }
+
+    @Override
+    protected void performOnReplicaProxy(
+        ReplicationProxyRequest<ReplicaRequest> proxyRequest,
+        ReplicationMode replicationMode,
+        BiConsumer<
+            Consumer<ActionListener<ReplicationOperation.ReplicaResponse>>,
+            ReplicationProxyRequest<ReplicaRequest>> performOnReplicasProxyBiConsumer
+    ) {
+        if (replicationMode == ReplicationMode.FULL_REPLICATION) {
```

Comment:
Isn't this always guaranteed to be FULL_REPLICATION?

Reply:
Yes, this will be FULL_REPLICATION always when the control reaches here. I can remove this.

Reply:
Add an assert

Reply:
ack